File: easy/0884_uncommon_words_from_two_sentences/solution.cpp
#include <sstream>
#include <unordered_map>
#include <vector>

class Solution {
public:
  std::vector<std::string> uncommonFromSentences(std::string s1,
                                                 std::string s2) {
    std::vector<std::string> uncommonWords;

    std::unordered_map<std::string, int> s1Counter, s2Counter;

    std::vector<std::string> words;
    std::string word;

    for (const std::string &s : std::vector<std::string>{s1, s2}) {
      std::stringstream iss(s);

      while (getline(iss, word, ' '))
        words.push_back(word);

      for (const std::string &word : words)
        s == s1 ? ++s1Counter[word] : ++s2Counter[word];

      words.clear();
    }

    for (const auto &[word, count] : s1Counter) {
      if (count == 1 && s2Counter.find(word) == s2Counter.end())
        uncommonWords.push_back(word);
    }

    for (const auto &[word, count] : s2Counter) {
      if (count == 1 && s1Counter.find(word) == s1Counter.end())
        uncommonWords.push_back(word);
    }

    return uncommonWords;
  }
};

File: easy/0884_uncommon_words_from_two_sentences/solution.py
from collections import Counter
from typing import List


class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        uncommon_words = []

        s1_counter = Counter(s1.split())
        s2_counter = Counter(s2.split())

        for word, count in s1_counter.items():
            if count == 1 and word not in s2_counter:
                uncommon_words.append(word)

        for word, count in s2_counter.items():
            if count == 1 and word not in s1_counter:
                uncommon_words.append(word)

        return uncommon_words
